src/lib/common/string.cpp

```diff
@@ -345,6 +405,21 @@ bool parseUrl(const std::string& url, std::string& host, int& port, std::string&
     }
   }
 
+  //
+  // Is 'port' a valid number?
+  // Remember the port is really an 'unsigned short', so numbers over MAXSHORT (0xFFFF) are
+  // not valid. Negative port numbers cannot exist and 0 is reserved.
+  //
+  if ((port > MAXSHORT) || (port <= 0))
```

Comment:
Ah, if MAX_PORT exists (probably defined to MAXSHORT :-)) then I will use it, of course.

Zero on the other hand I prefer not to let pass, as it seems to be a reserved value and cannot be used.
See https://en.wikipedia.org/wiki/List_of_TCP_and_UDP_port_numbers

[ Also, conveniently enough: atoi("invalid string") == 0 ]